packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -534,6 +542,12 @@ export class EntityService {
 
     const entity = await this.acquireLock(args, user);
 
+    if (args.data.name?.toLowerCase() === args.data.pluralDisplayName?.toLowerCase()) {
+      throw new AmplicationError(
+        `The entity name must be in the singular form.`
+      );
+    } 
```

Comment:
The user may choose to update only a single field (e.g. only the plural display name) so instead of checking against args. data you should also validate against the saved values

Reply:
np, where can I get access to the saved values?